src/Microsoft.ML.PipelineInference/PipelineSweeperSupportedMetrics.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using Microsoft.ML.Runtime.CommandLine;
+using Microsoft.ML.Runtime.EntryPoints;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.PipelineInference;
+using Microsoft.ML.Runtime.EntryPoints.JsonUtils;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.ML.Runtime.PipelineInference
+{
+    /// <summary>
+    /// PipelineSweeper will support metrics as they are added here.
+    /// </summary>
+    public sealed class PipelineSweeperSupportedMetrics
+    {
+        public enum Metrics
+        {
+            Auc,
+            AccuracyMicro,
+            AccuracyMacro,
+            L1,
+            L2,
+            F1,
+            AuPrc,
+            TopKAccuracy,
+            Rms,
+            LossFn,
+            RSquared,
+            LogLoss,
+            LogLossReduction,
+            Ndcg,
+            Dcg,
+            PositivePrecision,
+            PositiveRecall,
+            NegativePrecision,
+            NegativeRecall,
+            DrAtK,
+            DrAtPFpr,
+            DrAtNumPos,
+            NumAnomalies,
+            ThreshAtK,
+            ThreshAtP,
+            ThreshAtNumPos,
+            Nmi,
+            AvgMinScore,
+            Dbi
+        };
+
+        /// <summary>
+        /// Mapp Enum Metrics to a SupportedMetric
+        /// </summary>
+        private static readonly Dictionary<string, SupportedMetric> _map = new Dictionary<string, SupportedMetric>
```

Comment:
> _map [](start = 67, length = 5)

if the only usage for this is to contain a list of known metrics, the numb might be enough. 
Take a look at Enum.Parse. You can move the construction of new SupportedMetric in the GetSupportedMetric, if the conversion succeeds.

Reply:
I think Senja meant that the conversion from the enum to a string is not needed (see my comment below on the GetSupportedMetric method).

---
In reply to: [203220227](https://github.com/dotnet/machinelearning/pull/539#discussion_r203220227) [](ancestors = 203220227,203217600)